spec/datadog/core/crashtracking/component_spec.rb

```diff
@@ -0,0 +1,351 @@
+            .and_return(agent_base_url)
+          expect(::Libdatadog).to receive(:ld_library_path)
+            .and_return(ld_library_path)
+          expect(::Libdatadog).to receive(:path_to_crashtracking_receiver_binary)
+            .and_return(path_to_crashtracking_receiver_binary)
+
+          expect(described_class.build(settings, agent_settings, logger: logger)).to be_nil
+        end
+      end
+
+      context 'when missing `ld_library_path`' do
+        let(:ld_library_path) { nil }
+
+        it 'returns nil' do
+          expect(Datadog::Core::Crashtracking::TagBuilder).to receive(:call).with(settings)
+            .and_return(tags)
+          expect(Datadog::Core::Crashtracking::AgentBaseUrl).to receive(:resolve).with(agent_settings)
+            .and_return(agent_base_url)
+          expect(::Libdatadog).to receive(:ld_library_path)
+            .and_return(ld_library_path)
+          expect(::Libdatadog).to receive(:path_to_crashtracking_receiver_binary)
+            .and_return(path_to_crashtracking_receiver_binary)
+
+          expect(described_class.build(settings, agent_settings, logger: logger)).to be_nil
+        end
+      end
+
+      context 'when missing `path_to_crashtracking_receiver_binary`' do
+        let(:path_to_crashtracking_receiver_binary) { nil }
+
+        it 'returns nil' do
+          expect(Datadog::Core::Crashtracking::TagBuilder).to receive(:call).with(settings)
+            .and_return(tags)
+          expect(Datadog::Core::Crashtracking::AgentBaseUrl).to receive(:resolve).with(agent_settings)
+            .and_return(agent_base_url)
+          expect(::Libdatadog).to receive(:ld_library_path)
+            .and_return(ld_library_path)
+          expect(::Libdatadog).to receive(:path_to_crashtracking_receiver_binary)
+            .and_return(path_to_crashtracking_receiver_binary)
+
+          expect(described_class.build(settings, agent_settings, logger: logger)).to be_nil
+        end
+      end
+    end
+
+    context 'instance methods' do
+      # No crash tracker process should still be running at the start of each testcase
+      around do |example|
+        wait_for { `pgrep -f libdatadog-crashtracking-receiver` }.to be_empty
+        example.run
+        wait_for { `pgrep -f libdatadog-crashtracking-receiver` }.to be_empty
+      end
+
+      describe '#start' do
+        context 'when _native_start_or_update_on_fork raises an exception' do
+          it 'logs the exception' do
+            logger = Logger.new($stdout)
+            crashtracker = build_crashtracker(logger: logger)
+
+            expect(described_class).to receive(:_native_start_or_update_on_fork) { raise 'Test failure' }
+            expect(logger).to receive(:error).with(/Failed to start crash tracking: Test failure/)
+
+            crashtracker.start
+          end
+        end
+
+        it 'starts the crash tracker' do
+          crashtracker = build_crashtracker
```

Comment:
Consider extracting this into a `subject`

Reply:
I am adopting:  https://thoughtbot.com/blog/lets-not

In essence: Reducing mysterious guest by making the test example self contained and avoid indirection with rspec's caching DSL. Not DRYing with extraction has the the tradeoff of increase verbosity.